### DISCUSSION TITLE
Dotnet 8 Support

### DISCUSSION THREAD

**Discussion by satishviswanathan** [RicoSuter/NSwag#4609] [Q&A]
@RicoSuter  I have been using Nswag for quite a while and thank you for the library.

We recently started migrating our services from dotnet 7 to dotnet 8 and found that NSwag.AspnetCore is not yet compatible with dotnet 8. Reaching out to check the plans to publish a new version compatible with dotnet 8?

**Comment by lahma** [comment]
There's v14 preview available on NuGet.

**Comment by KarolZirex** [comment]
Do you know maybe when would be 14.0.0-preview010 with support dotnet8 in npm ready?
For now in npm is 14.0.0-preview003 and nswag doesn't work for me :/

> Error converting value "Net80" to type 'NSwag.Commands.Runtime'. Path 'runtime', line 2, position 20.

**Reply by grzegorzurbanek** [reply]
Hi, any news on this subject yet?

**Comment by xC0dex** [comment]
We are also waiting for a stable release with .NET 8 support. In my opinion, it was a wrong decision to put the v14 breaking changes and the .NET 8 support in the same branch.

**Reply by KarolZirex** [reply]
I am very disappointed about how long we have to wait for .NET 8 support, and we still don't know when it's gonna be

**Reply by lahma** [reply]
Anyone feeling a bit privileged here?

**Reply by xC0dex** [reply]
Hey @lahma no, that was not my intention, sorry. I tried to provide some feedback. I think this discussion about the latest .NET support is nothing new, and maybe I'm entirely wrong, but I guess the current issues in the preview are not caused by the .NET 8 support but to the breaking changes.
Long story short: I believe that the whole discussion wouldn't be necessary if the breaking changes were made in a separate branch. However, that's only my opinion. Perhaps there were good reasons to do it all in one branch.

**Reply by lahma** [reply]
My comment was pointed more towards the earlier comment about how free software isn't being delivered on time.

I believe the breaking changes have been waiting to happen and as resources for development are scarse, it's hard to please everyone. Honestly the 13.x version had such a myriad of configurations that it was even pain to get it to compile. It can be a pain to consume the changes, but should improve the possibility of maintaining the complex setup.

If you need 13.x with net8.0 support you could just fork and apply commits that you need.

**Reply by krzyhan** [reply]
@lahma  @RicoSuter  I created fork from `v13.20` and added .net8 - can you create a branch for v13.21? therefore I could create a PR with changes into official repo. After that `v13.21` could be released?
What do you think about that?

In the meantime, for everyone who is searching for net8 support -> https://github.com/krzyhan/NSwag/releases/tag/v13.21.0

**Reply by keithklim** [reply]
Is there support for .net 8?

**Reply by xC0dex** [reply]
@keithklim Yes, there is .NET 8 support since v14.0.0

**Reply by r-bkr** [reply]
Has anything progressed? When will `v13.21` be released?
